web/src/narrow.js

```diff
@@ -487,20 +477,12 @@ export function activate(raw_terms, opts) {
                 force_rerender: false,
             };
 
-            if (unread.messages_read_in_narrow) {
-                // We read some unread messages in a narrow. Instead of going back to
-                // where we were before the narrow, go to our first unread message (or
-                // the bottom of the feed, if there are no unread messages).
-                id_info.final_select_id = message_lists.current.first_unread_message_id();
-            } else {
-                // We narrowed, but only backwards in time (ie no unread were read). Try
-                // to go back to exactly where we were before narrowing.
-                // We scroll the user back to exactly the offset from the selected
-                // message that they were at the time that they narrowed.
-                // TODO: Make this correctly handle the case of resizing while narrowed.
-                then_select_offset = message_lists.current.pre_narrow_offset;
-                id_info.final_select_id = message_lists.current.selected_id();
+            if (opts.then_select_id !== -1) {
+                // Restore user's last position in narrow if user is navigation via browser back / forward button.
+                id_info.final_select_id = opts.then_select_id;
+                then_select_offset = opts.then_select_offset;
             }
```

Comment:
This bit seems not quite correct. The workflow I looked at is:
- In "combined feed", use `s` to visit a conversation you've already read.
- Use `Esc` to get back to the combined feed. 

Previously, that would leave you at the same offset as you'd been at before, but now it takes you the first unread. I am not sure if the `unread.messages_read_in_narrow` variable is a thing we need to make that workflow behave as designed, but I think this may be important for some people's workflows involving scrolling up in the "Combined feed" to revisit things.

Back button seems to go OK.

Reply:
Added another commit to fix that.